modules/@apostrophecms/ui/ui/apos/components/AposCombo.vue

```diff
@@ -121,22 +162,108 @@ export default {
     this.boxResizeObserver.unobserve(this.$refs.select);
   },
   methods: {
+    onTypeheadInput(e) {
+      this.thInput = e.target.value;
+      this.$emit('search', this.thInput);
+    },
+    onTypeheadKey(e) {
+      const stop = () => {
+        e.preventDefault();
+        e.stopPropagation();
+      };
+      switch (e.key) {
+        case 'ArrowDown':
+          stop();
+          this.focusListItem();
+          break;
+
+        case 'ArrowUp':
+          stop();
+          this.focusListItem(true);
+          break;
+
+        case 'Enter':
+          stop();
+          this.selectOption(this.options[this.focusedItemIndex]);
+          break;
+
+        case 'Tab':
+          stop();
+          this.closeList(null, true);
+          break;
+      }
+    },
     toggleList() {
       if (this.field.readOnly) {
         return;
       }
       this.showedList = !this.showedList;
 
       if (this.showedList) {
+        this.$emit('toggle', true);
         this.$nextTick(() => {
-          this.$refs.list.focus();
           this.focusedItemIndex = 0;
+          if (this.typehead) {
+            this.$refs.input.focus();
+          } else {
+            this.$refs.list.focus();
+          }
         });
       } else {
         this.$refs.select.focus();
         this.resetList();
       }
     },
+    listKeyDown(e) {
+      const stop = () => {
+        e.preventDefault();
+        e.stopPropagation();
+      };
+      switch (e.key) {
+        case ' ': // Brave issues
+        case 'Space': {
+          if (!this.typehead) {
+            e.preventDefault();
+            e.stopPropagation();
```

Comment:
Duplicated code, those two lines could be removed since the `stop` function already does the job:
```javascript
     e.preventDefault();
     e.stopPropagation();
```